src/Waives.Pipelines/Pipeline.cs

```diff
@@ -187,15 +180,7 @@ public Pipeline Then(Func<WaivesDocument, Task> action)
         /// <returns>The modified <see cref="Pipeline"/>.</returns>
         public Pipeline OnPipelineCompleted(Action action)
         {
-            var userAction = action ?? (() => { });
-
-            var previousAction = _onPipelineCompleted;
-            _onPipelineCompleted = () =>
-            {
-                previousAction();
-                userAction();
-            };
-
+            _onPipelineCompletedUserAction = action ?? (() => { });
```

Comment:
✋ We've removed the ability to register multiple `OnPipelineCompleted` handlers here. Was that intended? Should we guard against calling it more than once?